### PR TITLE
Update mypy to version 0.981

### PR DESCRIPTION
**Pull request by lucianopaz**
**What is this PR about?**

Update mypy version to 0.981. According to [this](https://github.com/python/mypy/issues/13627), it should fix the failed CI encountered in both #6151 and #6147 

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## Bugfixes / New features
- ...

## Docs / Maintenance
- ...

### CODE REVIEW

**Comment by maresb** [comment]
Is the other mypy warning new as of 0.981?

Perhaps the return type should be `Any` instead of `T`?

**Comment by maresb** [comment]
Ah, or should `DictToArrayBijection` be a generic on `T`?

EDIT: I got confused here, this should instead refer to `Compose`.

**Comment by lucianopaz** [comment]
Looks like returning `Any` did the trick.

I ran pre-commit on all the files and that was the only new warning that popped up.

File: pymc/blocking.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def __init__(self, fa: Callable[[PointType], T], fb: Callable[[RaveledVars], Poi
         self.fa = fa
         self.fb = fb
 
-    def __call__(self, x: RaveledVars) -> T:
+    def __call__(self, x: RaveledVars) -> Any:
```
Comment (maresb) [inline pymc/blocking.py]:
```suggestion
    def __call__(self, x: RaveledVars) -> T:
```

Comment (maresb) [reply pymc/blocking.py]: I think you should change `class Compose:` above to `class Compose(Generic[T]):`. This way, the `T` variable from `__init__` and `__call__` will be linked, and the type variable will actually define a relation.

**Review by maresb — CHANGES_REQUESTED** [review]
I'm thinking that `Compose` should be a generic

**Comment by lucianopaz** [comment]
> I'm thinking that `Compose` should be a generic

Thanks @maresb! The `Generic[T]` class annotation fixed the type hints

**Review by maresb — APPROVED** [review]

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**